src/styles/components/filter-bar.less

```diff
@@ -0,0 +1,116 @@
+.filter-bar {
+
+  display: flex;
+  flex-wrap: wrap-reverse;
+  width: 100%;
+
+  .filter-bar-left, .filter-bar-right {
+
+    display: inline-flex;
+    flex: 0 0 auto;
+    flex-wrap: wrap;
+    max-width: 100%;
+    width: auto;
+
+    .filter-bar-item {
+
+      flex: 0 0 auto;
+      max-width: 100%;
+
+    }
+
+  }
+
+  .filter-bar-right {
+
+    margin-left: auto;
+
+    &:last-child {
+
+      padding-right: 0;
+
+    }
+
+    .filter-bar-item:last-child {
+
+      // This overrides all of the specificity applied by the classes
+      // below.
+      padding-right: 0 !important;
```

Comment:
Interesting finding: filter search input doesn't expand to full width on Safari and FF at mobile view breakpoint:

![](http://cl.ly/3G0B2a2V0B3a/Image%202016-04-22%20at%2012.01.53%20PM.png)